lib/watchlist_edits.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+require_dependency "#{Rails.root}/lib/wiki_edits"
+class WatchlistEdits < WikiEdits
+  def initialize(wiki = nil, current_user= nil)
+    super(wiki)
+    @current_user = current_user
+  end
+
+  # Retrieves OAuth tokens for watchlist edits if the credentials are valid
+  def retrieve_tokens
```

Comment:
Since `watch_userpages` is the only method called by other code, this should be a private method.